client/client.go

```diff
@@ -316,11 +317,66 @@ func makeQueryString(data interface{}) (string, error) {
 	return "", fmt.Errorf("unsupported query string data: %#v", data)
 }
 
+// Remove all custom request serializer logic once APP-1331 is rolled out.
+type serializer func(any) ([]byte, error)
+
+func serializeQueryParamsToRequestBody(data any) ([]byte, error) {
+	m := make(map[string]any)
+	rv := reflect.ValueOf(data)
+	rt := reflect.TypeOf(data)
+	// If data is a map, just serialize it to JSON.
+	if rv.Kind() == reflect.Map {
+		return json.MarshalIndent(data, "", "  ")
+	}
+	for i := 0; i < rv.NumField(); i++ {
+		field := rv.Field(i)
+		tag := rt.Field(i).Tag.Get("url")
+		if tag == "" {
+			continue
+		}
+		value := field.Interface()
+		if reflect.DeepEqual(value, reflect.Zero(field.Type()).Interface()) {
```

Comment:
```suggestion
		if field.IsZero() {
```

no need for deep equal